extensions/azurePublishNew/src/node/azureResources/appInsights.ts

```diff
@@ -1,10 +1,19 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { ProvisionMethod, ProvisionWorkingSet, ResourceProvisionService } from '../types';
+import { ProvisionMethod, ProvisionWorkingSet, ResourceDefinition, ResourceProvisionService } from '../types';
 
+import { PAY_AS_YOU_GO_TIER, AZURE_HOSTING_GROUP_NAME } from './constants';
 import { AppInsightsConfigNew } from './types';
 
+export const appInsightsDefinition: ResourceDefinition = {
```

Comment:
I put the individual resource definitions in the same module as their provision method because they should be paired.  Callers will always need to know about both the definition and the service that provisions it.